File: app/src/main/java/com/example/yachae_sqlite/CustomCalendarActivity.java
package com.example.calendar;

import android.annotation.SuppressLint;
import android.app.AlertDialog;
import android.app.TimePickerDialog;
import android.content.Context;
import android.database.sqlite.SQLiteDatabase;
import android.graphics.drawable.GradientDrawable;
import android.provider.CalendarContract;
import android.util.AttributeSet;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.Button;
import android.widget.EditText;
import android.widget.GridView;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;
import android.widget.TimePicker;
import android.widget.Toast;

import androidx.annotation.Nullable;
import androidx.core.content.ContextCompat;

import com.google.android.material.chip.Chip;

import java.text.SimpleDateFormat;
import java.util.ArrayList;
import java.util.Calendar;
import java.util.Date;
import java.util.List;
import java.util.Locale;
import java.util.TimeZone;

public class CustomCalendarActivity extends LinearLayout {
    Button NextButton,PreviousButton;
    TextView CurrentDate;
    GridView gridView;
    private static final int MAX_CALENDAR_DAYS=42;
    Calendar calendar=Calendar.getInstance(Locale.KOREAN);
    Context context;
    SimpleDateFormat dateFormat= new SimpleDateFormat("yyyy MMMM dd",Locale.KOREAN);
    SimpleDateFormat monthFormat= new SimpleDateFormat("MMMM",Locale.KOREAN);
    SimpleDateFormat yearFormat= new SimpleDateFormat("yyyy",Locale.KOREAN);
    Chip chip_milk,chip_dairyproduct,chip_vegetable,chip_egg,chip_fish,chip_meat;

    MyGridAdapter myGridAdapter;
    AlertDialog alertDialog;
    List<Date> dates= new ArrayList<>();
    List<Events> eventsList= new ArrayList<>();
    DBOpenHelper dbOpenHelper;

    public CustomCalendarActivity(Context context) {
        super(context);
    }
    public CustomCalendarActivity(Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
        this.context=context;
        IntializeLayout();
        SetUpCalendar();

        PreviousButton.setOnClickListener(new OnClickListener() {
            @Override
            public void onClick(View view) {
                calendar.add(Calendar.MONTH,-1);
                SetUpCalendar();
            }
        });

        NextButton.setOnClickListener(new OnClickListener() {
            @Override
            public void onClick(View view) {
                calendar.add(Calendar.MONTH,1);
                SetUpCalendar();
            }
        });

        gridView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @SuppressLint("ResourceType")
            @Override
            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
                AlertDialog.Builder builder=new AlertDialog.Builder(context);
                View addView=LayoutInflater.from(parent.getContext()).inflate(R.layout.add_newevent_layout,null);
               // EditText EventName=addView.findViewById(R.id.events_id);
                //TextView EventTime=addView.findViewById(R.id.eventtime);

                Button AddEvent=addView.findViewById(R.id.addevent);
                builder.setView(addView);
                alertDialog=builder.create();
                alertDialog.show();

                AddEvent.setOnClickListener(new OnClickListener()
                {
                    @Override
                    public void onClick(View v){
                       //SaveEvent(EventName.getText().toString(),EventName.getText().toString());
                        // SetUpCalendar();
                        alertDialog.dismiss();
                       view.setBackgroundResource(R.drawable.stamp3black);
                    }
                });
            }
        });
    }


    public CustomCalendarActivity(Context context, @Nullable AttributeSet attrs, int defStyleAttr){
        super(context,attrs,defStyleAttr);
    }

    private void SaveEvent(String event, String date,String month,String year){
        dbOpenHelper=new DBOpenHelper(context);
        SQLiteDatabase database=dbOpenHelper.getWritableDatabase();
        dbOpenHelper.SaveEvent(event,date,month,year,database);
    }

    private void IntializeLayout(){
        LayoutInflater inflater=(LayoutInflater)context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
        View view=inflater.inflate(R.layout.calendar_layout,this);
        PreviousButton=view.findViewById(R.id.previousBtn);
        NextButton=view.findViewById(R.id.nextBtn);
        CurrentDate=view.findViewById(R.id.currnet_Date);
        gridView=view.findViewById(R.id.gridview);

        chip_milk=findViewById(R.id.chip_milk);
        chip_dairyproduct=findViewById(R.id.chip_dairyproduct);
        chip_vegetable=findViewById(R.id.chip_vegetable);
        chip_egg=findViewById(R.id.chip_egg);
        chip_fish=findViewById(R.id.chip_fish);
        chip_meat=findViewById(R.id.chip_meat);
    }

    private void SetUpCalendar(){
       String currentDate=dateFormat.format(calendar.getTime());
        CurrentDate.setText(currentDate);
        dates.clear();
        Calendar monthCalendar=(Calendar) calendar.clone();
        monthCalendar.set(Calendar.DAY_OF_MONTH,1);
        int FirstDayofMonth=monthCalendar.get(Calendar.DAY_OF_WEEK)-1;
        monthCalendar.add(Calendar.DAY_OF_MONTH,-FirstDayofMonth);

        while (dates.size()<MAX_CALENDAR_DAYS){
            dates.add(monthCalendar.getTime());
            monthCalendar.add(Calendar.DAY_OF_MONTH,1);
        }

        myGridAdapter=new MyGridAdapter(context,dates,calendar,eventsList);
        gridView.setAdapter(myGridAdapter);
    }
}

File: app/src/main/java/com/example/yachae_sqlite/CustomCalendarActivity.kt
package com.example.yachae_sqlite

import com.example.yachae_sqlite.MyGridAdapter
import com.google.android.material.chip.ChipGroup
import com.example.yachae_sqlite.DBOpenHelper
import android.database.sqlite.SQLiteDatabase
import android.widget.AdapterView.OnItemClickListener
import android.annotation.SuppressLint
import android.app.AlertDialog
import android.content.Context
import android.view.LayoutInflater
import com.example.yachae_sqlite.R
import com.google.android.material.chip.Chip
import androidx.annotation.RequiresApi
import android.os.Build
import android.util.AttributeSet
import android.util.Log
import android.view.View
import android.widget.*
import com.example.yachae_sqlite.CustomCalendarActivity
import java.text.SimpleDateFormat
import java.util.*

class CustomCalendarActivity : LinearLayout {
    lateinit var NextButton: Button
    lateinit var PreviousButton: Button
    lateinit var CurrentDate: TextView
    lateinit var gridView: GridView
    var calendar = Calendar.getInstance(Locale.KOREAN)
    //var context: Context? = null
    var dateFormat = SimpleDateFormat("yyyy MMMM", Locale.KOREAN)
    lateinit var myGridAdapter: MyGridAdapter
    lateinit var alertDialog: AlertDialog
    var dates: MutableList<Date> = ArrayList()
    var insert = false
    lateinit var chipName: String
    lateinit var chipGroup: ChipGroup
    lateinit  var dbopenhelper: DBOpenHelper
    lateinit var db: SQLiteDatabase

    constructor(context: Context?) : super(context) {}
    constructor(context: Context?, attrs: AttributeSet?) : super(context, attrs) {
        IntializeLayout()
        SetUpCalendar()
        PreviousButton!!.setOnClickListener { //-1한 월을 넣어줌
            calendar.add(Calendar.MONTH, -1)
            SetUpCalendar()
        }
        NextButton!!.setOnClickListener { //+1한 월을 넣어줌
            calendar.add(Calendar.MONTH, 1)
            SetUpCalendar()
        }
        dbopenhelper = DBOpenHelper(this@CustomCalendarActivity, "chipName", null, 1)
        //dbopenhelper.onCreate(db);
        gridView!!.onItemClickListener = OnItemClickListener { parent, view, position, id ->
            val builder = AlertDialog.Builder(context)
            val addView =
                LayoutInflater.from(parent.context).inflate(R.layout.add_newevent_layout, null)
            Log.d("Gridview", "Gridview 눌렸음")
            val AddEvent = addView.findViewById<Button>(R.id.addevent)
            val btn_close = addView.findViewById<Button>(R.id.btn_close_dialog)
            val chipgroup = addView.findViewById<ChipGroup>(R.id.chipgroup)
            val chip_milk = addView.findViewById<Chip>(R.id.chip_milk)
            val chip_dairyproduct = addView.findViewById<Chip>(R.id.chip_dairyproduct)
            val chip_vegetable = addView.findViewById<Chip>(R.id.chip_vegetable)
            val chip_egg = addView.findViewById<Chip>(R.id.chip_egg)
            val chip_fish = addView.findViewById<Chip>(R.id.chip_fish)
            val chip_meat = addView.findViewById<Chip>(R.id.chip_meat)
            val chip_fowls = addView.findViewById<Chip>(R.id.chip_fowls)
            builder.setView(addView)
            alertDialog = builder.create()
            alertDialog.show()
            AddEvent.isClickable = false
            chipName = chip_milk.text.toString()

            /*   chip_milk.setOnClickListener(new OnClickListener()
                    {
                        @Override
                        public void onClick(View v){
                            if(chip_milk.isChecked()==true){
                            Log.d("chip_milk","chip_milk 성공");
                            insert=true;
                            //insert=dbOpenHelper.insertData(chipName);
                            //텍스트로 insert}
                        }
                    });*/AddEvent.setOnClickListener(object : OnClickListener {
            var str_name = ""
            override fun onClick(v: View) {
                Log.d("기록하기 버튼", "눌렸습니다.")
                //SaveEvent(EventName.getText().toString(),EventName.getText().toString());
                if (chipgroup.checkedChipId == R.id.chip_milk) {
                    Log.d("CHIP_MILK", "milk 눌림")
                    str_name = chip_milk.text.toString()
                }
                //db = dbopenhelper!!.writableDatabase
                //db.execSQL("INSERT INTO chipName VALUES('$str_name');")
                //db.close()
                view.setBackgroundResource(R.drawable.stamp3black)
                alertDialog.dismiss()
            }
        })
            btn_close.setOnClickListener {
                Log.d("CLOSEBTN", "취소 버튼 눌림")
                alertDialog.dismiss()
            }
        }
    }

    @RequiresApi(api = Build.VERSION_CODES.HONEYCOMB)
    constructor(context: Context?, attrs: AttributeSet?, defStyleAttr: Int) : super(
        context,
        attrs,
        defStyleAttr
    ) {
    }

    /*private void SaveEvent(String event, String date,String month,String year){
        dbOpenHelper=new DBOpenHelper(context);
        SQLiteDatabase database=dbOpenHelper.getWritableDatabase();
        dbOpenHelper.SaveEvent(event,date,month,year,database);
    }*/
    private fun IntializeLayout() {
        val inflater = context!!.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
        val view = inflater.inflate(R.layout.calendar_layout, this)
        PreviousButton = view.findViewById(R.id.previousBtn)
        NextButton = view.findViewById(R.id.nextBtn)
        CurrentDate = view.findViewById(R.id.currnet_Date)
        gridView = view.findViewById(R.id.gridview)
    }

    private fun SetUpCalendar() {
        val currentDate = dateFormat.format(calendar.time)
        CurrentDate!!.text = currentDate
        dates.clear()
        //날짜 복사해서 변수 생성
        val monthCalendar = calendar.clone() as Calendar
        //1일로 셋팅
        monthCalendar[Calendar.DAY_OF_MONTH] = 1
        //요일 가져와서 -1=>일요일=1, 월요일=2...
        val FirstDayofMonth = monthCalendar[Calendar.DAY_OF_WEEK] - 1
        //날짜 셋팅
        monthCalendar.add(Calendar.DAY_OF_MONTH, -FirstDayofMonth)
        while (dates.size < MAX_CALENDAR_DAYS) {
            //리스트에 날짜 등록
            dates.add(monthCalendar.time)
            //1일씩 늘린 날짜로 변경
            monthCalendar.add(Calendar.DAY_OF_MONTH, 1)
        }
        myGridAdapter = MyGridAdapter(context!!, dates, calendar)
        gridView!!.adapter = myGridAdapter
    }

    companion object {
        private const val MAX_CALENDAR_DAYS = 42
    }
}